Source/WebCore/history/BackForwardCache.cpp

```diff
@@ -535,14 +541,18 @@ std::unique_ptr<CachedPage> BackForwardCache::suspendPage(Page& page)
 
 std::unique_ptr<CachedPage> BackForwardCache::take(HistoryItem& item, Page* page)
 {
-    if (!item.m_cachedPage) {
-        if (item.m_pruningReason != PruningReason::None)
-            logBackForwardCacheFailureDiagnosticMessage(page, pruningReasonToDiagnosticLoggingKey(item.m_pruningReason));
+    auto it = m_cachedPageMap.find(item.identifier());
+    if (it == m_cachedPageMap.end())
+        return nullptr;
+    if (auto* pruningReason = std::get_if<PruningReason>(&it->value)) {
+        if (*pruningReason != PruningReason::None)
+            logBackForwardCacheFailureDiagnosticMessage(page, pruningReasonToDiagnosticLoggingKey(*pruningReason));
         return nullptr;
     }
 
-    m_items.remove(&item);
-    auto cachedPage = item.takeCachedPage();
+    m_items.remove(item.identifier());
+    auto cachedPage = std::get<UniqueRef<CachedPage>>(m_cachedPageMap.take(it));
```

Comment:
WTF::switchOn would be more future-proof than std::get_if followed by std::get.  If someone adds a third variant alternative then it will be a compiler error.